internal/versions/rest_api_versions.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+	"gorm.io/gorm"
+)
+
+type restAPIVersionsHandler struct {
+	log                      logrus.FieldLogger
+	releaseHandler           oc.Release
+	mustGatherVersions       MustGatherVersions
+	ignoredOpenshiftVersions []string
+	db                       *gorm.DB
+}
+
+// GetReleaseImage retrieves a release image based on a specified OpenShift version and CPU architecture.
+// If the provided OpenShift version includes a patch version, it attempts to retrieve an exact match for the release image if available.
+// For OpenShift versions specified as major.minor, it fetches the latest matching release image.
+// The function returns an error Returns an error for other formats of OpenShift version or if no matching image can be found.
+func (h *restAPIVersionsHandler) GetReleaseImage(_ context.Context, openshiftVersion, cpuArchitecture, _ string) (*models.ReleaseImage, error) {
+	// validations
+	if err := validateCPUArchitecture(cpuArchitecture); err != nil {
+		return nil, err
+	}
+	versionFormat := common.GetVersionFormat(openshiftVersion)
+	if versionFormat == common.NoneVersion || versionFormat == common.MajorVersion {
+		return nil,
+			errors.Errorf(
+				"invalid openshiftVersion '%s'. Expected format: 'major.minor' or 'major.minor.patch', optionally followed by a prerelease identifier",
+				openshiftVersion,
+			)
+	}
+
+	query := h.db.Session(&gorm.Session{})
+	cleanQuery := h.db.Session(&gorm.Session{})
+	query = query.Where(&models.ReleaseImage{})
+	if cpuArchitecture == common.MultiCPUArchitecture {
+		query = query.Where("cardinality(cpu_architectures) > 1")
+	} else {
+		query = query.Where(cleanQuery.Where("? = ANY(cpu_architectures)", cpuArchitecture).Or("cpu_architecture = ?", cpuArchitecture))
+	}
+	newVersion := openshiftVersion
+
+	// Find the exact version
+	if versionFormat == common.MajorMinorPatchVersion {
+		// add multi suffix if missing
+		if cpuArchitecture == common.MultiCPUArchitecture && !strings.HasSuffix(newVersion, "-multi") {
+			newVersion = newVersion + "-multi"
+		}
+		query = query.Where("version = ?", newVersion)
+		var releaseImage models.ReleaseImage
+		err := query.Take(&releaseImage).Error
+		if err != nil {
+			if errors.Is(err, gorm.ErrRecordNotFound) {
+				return nil, errors.Errorf("no release image found for openshiftVersion: '%s' and CPU architecture '%s'", openshiftVersion, cpuArchitecture)
+			}
+
+			return nil, err
+		}
+
+		if err = assertNotIgnored(*releaseImage.Version, h.ignoredOpenshiftVersions); err != nil {
+			return nil, err
+		}
+
+		return &releaseImage, err
+	}
+
+	// openshiftVersion must be major.minor, find the latest version matching
+
+	// In case openshiftVersion is in the format of major.minor-multi
+	newVersion = strings.TrimSuffix(openshiftVersion, "-multi")
+
+	// Escape SQL wildcards in case thay are approved as a version
+	newVersion = common.EscapePlaceHolders(newVersion)
```

Comment:
Instead of escaping the version and use sql, won't it be simpler to find relevant release images with something like gofunk? E.g. https://github.com/openshift/assisted-service/blob/cc56ded8ef0a9dc3f95e2c8a65d08e96546cd72b/internal/versions/versions.go#L175